src/lib.rs

```diff
@@ -465,24 +477,30 @@ fn pan_orbit_camera(
                 has_moved = true;
             }
         } else if scroll.abs() > 0.0 {
-            let apply_scroll = |value: f32| value - scroll * value * 0.2;
+            let delta = - scroll * pan_orbit.target_radius * 0.2;
+            pan_orbit.target_radius = util::apply_zoom_limits(
+                pan_orbit.target_radius + delta,
+                pan_orbit.zoom_upper_limit,
+                pan_orbit.zoom_lower_limit,
+            );
 
-            if let Projection::Orthographic(ref mut p) = *projection {
-                p.scale = util::apply_zoom_limits(
-                    apply_scroll(p.scale),
-                    pan_orbit.zoom_upper_limit,
-                    pan_orbit.zoom_lower_limit,
-                );
-            } else {
-                pan_orbit.radius = pan_orbit.radius.map(|radius| {
-                    util::apply_zoom_limits(
-                        apply_scroll(radius),
-                        pan_orbit.zoom_upper_limit,
-                        pan_orbit.zoom_lower_limit,
-                    )
-                });
+        }
+        if let Projection::Orthographic(ref mut p) = *projection {
+            let target = p.scale.lerp(&pan_orbit.target_radius, &(1.0 - pan_orbit.zoom_smoothness));
+            if (target - p.scale).abs() > 0.001 {
```

Comment:
You are only moving towards the target if the distance is greater. This means technically you'll never actually reach the target, which I think is not ideal.

Let's follow the pattern of alpha/beta, and snap to the target if the distance is less than `0.001`, to guarantee we'll always reach the target.